deirokay/statements/builtin/contain.py

```diff
@@ -386,9 +353,68 @@ def _generate_report(self, analysis):
                 else values_report
                 if self.report_limit is None
                 else values_report[: self.report_limit]
-            )
+            ),
+            "rule_report": rule_report
         }
 
+    def _create_values_report(self, analysis):
+        columns = [analysis[col] for col in analysis.columns[:-4]]
+        serialized = (
+            treater.serialize(column) for treater, column in zip(self.treaters, columns)
+        )
+        rows = zip(*(s["values"] for s in serialized))
+        rows = (list(row) for row in rows)
+        values_report = [
+            {
+                "value": value_row,
+                "count": analysis_row.count,
+                "result": analysis_row.occurrences_result,
+            }
+            for value_row, analysis_row in zip(rows, analysis.itertuples())
+        ]
+        return values_report
+
+    def _create_rule_report(self, analysis):
+        values_in_df = analysis[(analysis["count"] > 0)]
+        perc_in_values = float(values_in_df["in_values"].mean() * 100)
+
+        in_values = analysis[(analysis["in_values"]) & (analysis["max"] > 0)]
+        if len(in_values):
+            perc_values_in_df = float((in_values["count"] > 0).mean()) * 100
+        else:
+            perc_values_in_df = 0
+
+        if self.rule == "only":
+            result = perc_in_values >= 100 - self.allowed_perc_error
+            return {
+                    "rule": self.rule,
+                    "perc_in_values": perc_in_values,
+                    "allowed_perc_error": self.allowed_perc_error,
+                    "result": result,
+                }
+        elif self.rule == "all":
+            in_values = analysis[(analysis["in_values"]) & (analysis["max"] > 0)]
+            if len(in_values):
+                result = perc_values_in_df >= 100 - self.allowed_perc_error
+                return {
+                        "rule": self.rule,
+                        "perc_values_in_df": perc_values_in_df,
+                        "allowed_perc_error": self.allowed_perc_error,
+                        "result": result,
+                    }
+            else:
+                return {
+                    "rule": self.rule,
+                    "result": True  # All the values have check of not contain 
+                }
+        elif self.rule == "all_and_only":
+            return {
+                "rule": self.rule,
+                "perc_in_values": perc_in_values,
+                "perc_values_in_df": perc_values_in_df,
+                "result": perc_in_values == 100 and perc_values_in_df == 100
+            }
```

Comment:
shouldn't you assert in `init` that the user will never use "allowed_perc_error != 0" and "rule = all_and_only" ?

Reply:
True. I'll add the check on init